### DISCUSSION TITLE
Calling refprop from coolprop in C++ (not python)

### DISCUSSION THREAD

**Discussion by eliesaikali** [CoolProp/CoolProp#2231] [Q&A]
Hello !

I am trying to call refprop from coolprop in my C++ interface, but I am having issues (shared library of refprop not found)

I am working on Mac (arm64) and using a static library of coolprop
I managed to compile a 9.1 refprop shared library (librefprop.dylib) using the github project REFPROP-cmake

Although I added the path towards librefprop.dylib in LD_LIBRARY_PATH & PATH, it seems that coolprop is always looking for the library  in /opt/refprop/ ... Any idea on how to solve this ?

I also tried to export the var ALTERNATIVE_REFPROP_PATH with the good path but I am getting the same issue.

Thanks !

**Comment by ibell** [comment]
Please read the docs relating to setting the configuration variables for
refprop linkage.

**Reply by eliesaikali** [reply]
As I said, I want this to work in C++ and not python

If you are talking about what is documented here (http://www.coolprop.org/coolprop/REFPROP.html),
surely I have read it ! Again, for me, this works well for python but not in C++  ...

Thanks

**Reply by ibell** [reply]
The C++ interface is a more or less one-to-one mapping of the Python one, so you would need to set configuration variables in C++ like you do in Python, like in the docs you linked to. Same functions, same data structures, only difference is the namespaces.

**Reply by eliesaikali** [reply]
Ok Thanks Prof Bell for your help.

Last question, I managed to compile the 9.1 version of refprop (using the REFPROP-cmake project) and I managed finally to call it via coolprop :) Hovever for some functions, I have this error:

PHI0dll function is not available in your version of REFPROP. Please upgrade ...


The error is clear and I should compile a newer version of RefProp.


I am trying with 9.11, but I dont think the REFPROP-cmake project is up to date for this version ? In fact, I see that I dont have PASS_FTN.FOR file in the new version ! Is this normal ?

Thanks for your help

**Reply by ibell** [reply]
You need version 10 for PHI0dll function

**Reply by eliesaikali** [reply]
So version 11 is not yet supported ? I should down-grade ?

**Reply by eliesaikali** [reply]
Sorry for the confusion, In fact its 10 not 11 ...
But the problem is that I dont have the file PASS_FTN.FOR in the 10 version ...

**Reply by ibell** [reply]
It is in the DLLFILES folder in version 10

**Reply by ibell** [reply]
which is inside the FORTRAN folder

**Reply by ibell** [reply]
Make sure you are doing a clean rebuild if you previously built with version 9
